### PR TITLE
Fix core functional test for authenticated sources

### PR DESCRIPTION
**Pull request by spadapet**
I changed the test `Restore_WithAuthenticatedSource_Succeeds` to use specific sources for testing restore against authenticated sources. Before, all servers in the nuget.config file would be checked and they don't all work. Now we'll just check servers that we know work, and only one at a time.

@alpaix @zhili1208 @emgarten

### CODE REVIEW

**Comment by zhili1208** [comment]
:shipit:

File: test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs [hunk 1]
```diff
@@ -10,14 +11,31 @@ namespace NuGet.XPlat.FuncTest
...
+        [InlineData(TestServers.NuGetServer, false)]
+        [InlineData(TestServers.ProGet, false)]
+        // Try with config file in a different directory
+        [InlineData(TestServers.Artifactory, true)]
+        [InlineData(TestServers.Klondike, true)]
+        [InlineData(TestServers.MyGet, true)]
+        [InlineData(TestServers.Nexus, true)]
+        [InlineData(TestServers.NuGetServer, true)]
+        [InlineData(TestServers.ProGet, true)]
+        public void Restore_WithAuthenticatedSource_Succeeds(string sourceUri, bool configInDifferentDirectory)
```
Comment (alpaix) [inline test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs]: Only one server from this list is configured for authentication, NuGetServer. I think others should be removed.

File: test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs [hunk 2]
```diff
@@ -28,67 +46,40 @@ public void Restore_WithAuthenticatedSource_Succeeds()
...
-                    packagesDir,
-                    "--configfile",
-                    configFile,
-                    "--no-cache"
-                };
+                    args.Add("--configfile");
+                    args.Add(configFile);
+                }
+
+                System.Console.WriteLine(
```
Comment (spadapet) [inline test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs]: I think I'll get rid of this writeline, it was there to help me debug

File: test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs (hunk 1 above)
Comment (spadapet) [inline test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs]: I can rename the file and function, it's useful to try to restore against all of these servers.
Would that work?

---

In reply to: [63444223](https://github.com/NuGet/NuGet.Client/pull/596#discussion_r63444223) [](ancestors = 63444223)

**Comment by emgarten** [comment]
:shipit:

File: test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs (hunk 1 above)
Comment (alpaix) [inline test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs]: Sure please rename it.

**Pull request merged**